DataRepo/templatetags/customtags.py

```diff
@@ -220,3 +220,24 @@ def uniquify(retval, unused):
     differently in 2 parts of a conditional, but with the same ID.  Just supply a different value to unused.
     """
     return retval
+
+
+@register.simple_tag
+def get_manytomany_rec(recs, pk):
+    """
+    If a M:M ralted table is marked in compositeviews as a full_join, this method identifies the M:M related record
+    that belongs to the root table record, as if this was a proper SQL left join.  While django always returns every
+    related table record with every root table record, this method essentially simulates a proper join by only
+    selecting the join record that was involved in the original query.  It uses an annotated version of the M:M related
+    table record that was added to the root table record using getFullJoinAnnotations().
+
+    It returns a list in each case so that full join can be turned off and on by simply tiggling the `full_join`
+    boolean value in compositeviews.py.
+    """
+    if pk != "":
+        for rec in recs:
+            if rec.pk == pk:
+                return [rec]
+        return None
```

Comment:
One other minor comment. This looks like it returns the first matching record (where `rec.pk == pk`). That might be helpful in the docstring. Also, this sort of thing could be done using [`next`](https://docs.python.org/3/library/functions.html#next):
```suggestion
        next(([x] for x in recs if x.pk == pk), None)
```

Reply:
What's the advantage?  Is that more efficient?  Seems harder to read.

Maybe... This is a queryset - so can I use `.get()`?  Maybe that would be better inside a try.

- [ ] Add the behavior to the doc string - that it assumes distinct/unique primary keys in the queryset.

Reply:
Might also be nice to be able to manually set the `full_join` option via `performQuery`... Maybe I'll create a "nice-to-have" feature issue.

And... I don't know that `full_join` (not `split_row_counts` for that matter), as a key in the compositeviews data structures conveys what it's for very well.  Maybe I'll re-think that key name and/or add a comment or put it in the doc string.  I should also:

- [ ] Update the DEV doc to include the new manytomany, full_join, and split_row_counts